packages/react-devtools-shell/src/devtools.js

```diff
@@ -50,6 +54,10 @@ mountButton.addEventListener('click', function() {
   }
 });
 
+function hookNamesModuleLoaderFunction() {
```

Comment:
Ditto here

Reply:
what do you mean by asking the developer to do?

Reply:
Seems like right now, if someone needs to use the `react-devtools-inline` package with react hooks, they will also need to add a `hookNamesModuleLoaderFunction` and manually modify the webpack public path, which seems a bit unergonomic. This code seems like it should live in the react-devtools packages instead of in userland. I understand why it's here right now, but I wonder if it's the right solution.

Of course I might be misunderstanding something here!!!

Reply:
Also, if developers forget to add this, does this code fail gracefully or let the developer know somehow?

Reply:
> Also, if developers forget to add this, does this code fail gracefully or let the developer know somehow?

If they don't pass this prop, the feature gracefully degrades (same as if the feature flag were disabled):
https://github.com/bvaughn/react/blob/b312571db910fde319d87570d61719b276814d0b/packages/react-devtools-shared/src/devtools/views/Components/InspectedElementContext.js#L127-L128

> Seems like right now, if someone needs to use the `react-devtools-inline` package with react hooks, they will also need to add a `hookNamesModuleLoaderFunction` and manually modify the webpack public path, which seems a bit unergonomic.

Agreed! It's not great, but until/unless we figure the Webpack upgrade issue, I don't know of a better way to approach this.

Reply:
Makes sense. Maybe we could add this as a TODO fix once we upgrade webpack and update the documentation to reflect this then :)

Reply:
Sure! I added a few TODO comments in the various place this code is referenced.

Reply:
oh got it, yeah, that makes sense

Reply:
just curious, how often are people embedding the inline package somewhere else? but yeah if this isn't fixed with webpack 5 agree that it would be worth to update docs somewhere or at least create an issue to keep track